Lets explore a bit on spark UI.  
Spark UI stays up till our application is running.

In [1]:
from pyspark.sql import SparkSession

In [2]:
spark = SparkSession.builder.getOrCreate()

25/02/21 11:16:10 WARN Utils: Your hostname, lenovo resolves to a loopback address: 127.0.1.1; using 192.168.29.125 instead (on interface wlp3s0)
25/02/21 11:16:10 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
25/02/21 11:16:14 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
25/02/21 11:16:18 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
25/02/21 11:16:18 WARN Utils: Service 'SparkUI' could not bind on port 4041. Attempting port 4042.


ok, now we can see SPARK UI  
It is available at `http://localhost:4040/`  
But look at the warning message in the console, for me SPARK UI is available at 4042